src/Mod/PartDesign/App/FeatureHelix.cpp

```diff
@@ -0,0 +1,512 @@
+}
+
+
+TopoDS_Shape Helix::generateHelixPath(void)
+{
+    double pitch = Pitch.getValue();
+    double height = Height.getValue();
+    bool leftHanded = LeftHanded.getValue();
+    bool reversed = Reversed.getValue();
+    double angle = Angle.getValue();
+    if (angle < Precision::Confusion() && angle > -Precision::Confusion())
+        angle = 0.0;
+
+    // get revolve axis
+    Base::Vector3d b = Base.getValue();
+    gp_Pnt pnt(b.x,b.y,b.z);
+    Base::Vector3d v = Axis.getValue();
+    gp_Dir dir(v.x,v.y,v.z);
+
+    Base::Vector3d normal = getProfileNormal();
+    Base::Vector3d start = v.Cross(normal);  // pointing towards the desired helix start point.
+    gp_Dir dir_start(start.x, start.y, start.z);
+
+    // Determine radius as the minimum distance between sketchshape and axis.
+    // also find out in what quadrant relative to the axis the profile is located.
+    double radius = 1e99;
+    bool turned = false;
+    double startOffset = 1e99;
+    TopoDS_Shape sketchshape = getVerifiedFace();
+    BRepBuilderAPI_MakeEdge axisEdge(gp_Lin(pnt, dir));
+    BRepBuilderAPI_MakeEdge startEdge(gp_Lin(pnt, dir_start));
+    for (TopExp_Explorer xp(sketchshape, TopAbs_FACE); xp.More(); xp.Next()) {
+        const TopoDS_Face face =  TopoDS::Face(xp.Current());
+        TopoDS_Wire wire = ShapeAnalysis::OuterWire(face);
+        BRepExtrema_DistShapeShape distR(wire, axisEdge.Shape(), Precision::Confusion());
+        if (distR.IsDone()) {
+            if (distR.Value() < radius) {
+                radius = distR.Value();
+                const gp_Pnt p1 = distR.PointOnShape1(1);
+                const gp_Pnt p2 = distR.PointOnShape2(1);
+                double offsetProfile = p1.X()*dir_start.X() + p1.Y()*dir_start.Y() + p1.Z()*dir_start.Z();
+                double offsetAxis = p2.X()*dir_start.X() + p2.Y()*dir_start.Y() + p2.Z()*dir_start.Z();
+                turned = (offsetProfile < offsetAxis);
+            }
+        }
+        BRepExtrema_DistShapeShape distStart(wire, startEdge.Shape(), Precision::Confusion());
+        if (distStart.IsDone()) {
+            if (distStart.Value() < abs(startOffset)) {
+                const gp_Pnt p1 = distStart.PointOnShape1(1);
+                const gp_Pnt p2 = distStart.PointOnShape2(1);
+                double offsetProfile = p1.X()*dir.X() + p1.Y()*dir.Y() + p1.Z()*dir.Z();
+                double offsetAxis = p2.X()*dir.X() + p2.Y()*dir.Y() + p2.Z()*dir.Z();
+                startOffset = offsetProfile - offsetAxis;
+            }
+        }
+
+    }
+
+    if (radius <  Precision::Confusion()) {
+        // in this case ensure that axis is not in the sketch plane
+        if (v*normal < Precision::Confusion())
+            throw Base::ValueError("Error: Result is self intersecting");
+        radius = 1.0; //fallback to radius 1
+        startOffset = 0.0;
+    }
+
+    //build the helix path
+    TopoShape helix = helix.makeLongHelix(pitch, height, radius, angle, leftHanded);
```

Comment:
This is a weird construct and raises a compiler warning: variable helix is uninitialized when used within its own initialization [-Wuninitialized]

It should be replaced e.g. with: `TopoDS_Shape path = TopoShape().makeLongHelix(pitch, height, radius, angle, leftHanded);`

Reply:
I don't know what happened here. Maybe my baby boy was tapping on the keyboard when I was looking away :). Anyway it is fixed and commited now.